torchgeo/datasets/bright.py

```diff
@@ -0,0 +1,338 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""BRIGHT dataset."""
+
+import os
+import textwrap
+from collections.abc import Callable
+from typing import ClassVar
+
+import matplotlib.patches as mpatches
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from einops import repeat
+from matplotlib import colors
+from matplotlib.figure import Figure
+from torch import Tensor
+
+from .errors import DatasetNotFoundError
+from .geo import NonGeoDataset
+from .utils import Path, check_integrity, download_url, extract_archive
+
+
+class BRIGHTDFC2025(NonGeoDataset):
```

Comment:
Wondering if we should call it BRIGHT instead of BRIGHT DFC2025.

Reply:
Since this did not seem to be the final version, rather the Challenge version, I included it in the name. However, the paper and also comment below noted, that this is not the final version. Since, I would expect or hope that the dataset format stays the same for the final version, we could also add a version parameter.

Reply:
Alright, then let's assume we will have more BRIGHT datasets in the future.

Reply:
wait so do you mean keep the dataset name like it is and instead just put it under a "Bright" umbrella of possible datasets in the docs?

Reply:
Yep. So this dataset will be the DFC2025 version, and we may add more versions in the future. Same reason you used `bright.py` we should also use BRIGHT in the docs.